subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/state/UpToDateIntegTest.groovy

```diff
@@ -283,34 +283,16 @@ public abstract class CreateEmptyDirectory extends DefaultTask {
             }
         """
         def inputFile = file("input/input.txt")
-        inputFile.text = "[1] Hello World!"
-        def previousInputFileLength = inputFile.length()
+        inputFile.text = "[0] Hello World!"
+        def originalInputFileLength = inputFile.length()
 
-        when:
-        run "printFile"
-        then:
-        executedAndNotSkipped(":printFile")
-
-        when:
-        inputFile.text = "[2] Hello World!"
-        run "printFile"
-        then:
-        executedAndNotSkipped(":printFile")
-        previousInputFileLength == inputFile.length()
-
-        when:
-        inputFile.text = "[3] Hello World!"
-        run "printFile"
-        then:
-        executedAndNotSkipped(":printFile")
-        previousInputFileLength == inputFile.length()
-
-        when:
-        inputFile.text = "[4] Hello World!"
-        run "printFile"
-        then:
-        executedAndNotSkipped(":printFile")
-        previousInputFileLength == inputFile.length()
+        expect:
+        [1, 2, 3, 4].each {
```

Comment:
```suggestion
        (1..4).each {
```

You can use a Groovy range here.

Reply:
Nice shortcut!

I will also add `outputContains(givenText)` just to check that we print actual text